### DISCUSSION TITLE
v3.5.33-Beta

### DISCUSSION THREAD

**Discussion by ikb42** [LightwaveSmartHome/homeassistant-lightwave-smart#23] [Announcements]
Add "update" to manage firmware
Improve sensors and events

<hr /><em>This discussion was created from the release <a href='https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v3.5.33-Beta'>v3.5.33-Beta</a>.</em>

**Comment by xela1** [comment]
Hi @ikb42 

I've just installed this on my dev instance, I can see the firmware status, but it's on a second instance of the device with the same name

![image](https://github.com/user-attachments/assets/059bb561-d89a-469b-b67f-f213919c8cae)

![image](https://github.com/user-attachments/assets/85d0bf3f-c6b5-438e-a381-0c877318c040)

is this expected?

**Reply by ikb42** [reply]
Hello @xela1 - thanks for the feedback!  I'd like to get this version released soon, so this really helps.
Yes this is as expected as it stands, perhaps it can be improved on in future.

**Reply by xela1** [reply]
I'll run some tests on it to see if it all behaves as expected, I don't think any of my devices need updating at the moment so won't be able to test that just yet (unless there's a way to force a downgrade?)

**Reply by ikb42** [reply]
Excellent
Downgrade is not an option, but in terms of testing it's general / existing functionality behaving as expected that I'm keen to verify.
My goal is to get this integration HA approved - or "Works with Home Assistant" approved.

**Comment by nuttersrest** [comment]
Something seems to have gone awry with the upgrade on this.

I now have two LinkPlus Entries with one only having the Firmware and the second one having the rest of the info on the LinkPlus

![image](https://github.com/user-attachments/assets/24e0d8bd-55cc-434d-a450-338dab132360)

The switches are similar but also have broken my dashboards and automations as the underlying switch seems to have been renamed. I now have three entries for a switch. One tile has joined the origional two names of the switches, which includes the firmware and actual switches though the underlying switches were just called switch (making it hard to identify and I had to go around and rename everything again. The other two still retaining the origional names and have all the current/power/voltage information. 

![image](https://github.com/user-attachments/assets/fa354415-3508-4650-beba-6cd0c8f33a67)

![image](https://github.com/user-attachments/assets/850a99c6-1dff-40e9-b2e1-e700adc28ec3)

![image](https://github.com/user-attachments/assets/5bf86c98-1f12-4cb7-8d5e-0fb96f715d49)

Although when you update the firmware it impacts the two sockets on a dual socket. It would perhaps, if possible, make things more sensible to have the firmware button duplicated on both switch sockets leaving everything for that socket (switch, power readings and firmware) in one tile. If that is not possible just have the firmware on the first sockets tile.

Thanks for creating and supporting this integration.

**Reply by nuttersrest** [reply]
Just a small follow up on the above, the actual switches that are renamed revert to being called Switch when HomeAssistant is restarted. Any entity on a dashboard that has used that particular switch name, then reverts back to the parent name (which has the the original two names joined together) making it impossible to know which of the devices are being turned off or on.

**Reply by ashy-87** [reply]
I’m also seeing the same reverting devices on reboot, they lose their area attributes too. I get a handful of Gen1 devices that ‘reset’ on reboot too, losing their area, name and type. I hope to see a fix, is there anything I can offer to help identify the problem?

**Comment by Gazza-B** [comment]
Is anybody here aware that the latest versions of HA break the lightwave integration with TRV’s which happened to me , the tiles are just blank, restoring 24.12 fetches them back, another user reported it breaks the start/Stop relays.

**Comment by xela1** [comment]
Are you running the version from this repository or bigbadblunt's repository, if this one, which version, I'm on the latest HA but not the latest beta release of the integration and mine are all fine

![image](https://github.com/user-attachments/assets/f56167b6-c8f4-435f-bef1-7ede993b78e5)

**Comment by Gazza-B** [comment]
It was bigbadblunts... changed to later integration..works fine

Gary Bridge MIPS, MIET, MAES.


GB Broadcast Radio Technology.
________________________________

**Comment by AKWillows** [comment]
I am in the process of working through changing from the original integration to this one due to the functionality of the previous integration which was deprecated on HA 2025.1. I'm using the latest Beta of this integration 

However I am really struggling with the devices and entities that are created. I have the same issues as others have noted above, but also some strange behaviour with the devices and entities created for the L40MK2 plug-in sockets. Like most other things there are two devices, one has the switch and the firmware and the other has an Event (which always shows as Unknown) and the Configuration and diagnostic items.

<img width="668" alt="Screenshot 2025-02-02 at 17 48 57" src="https://github.com/user-attachments/assets/8c7f3efc-1a9b-47be-8bec-81064b0a80dc" />

<img width="686" alt="Screenshot 2025-02-02 at 17 49 11" src="https://github.com/user-attachments/assets/c0a8b2b2-5fe3-49e4-8238-a73c922393fd" />

This is different to the way it is set up for a L21MK3 light switch which has just the firmware on a second device: 
<img width="688" alt="Screenshot 2025-02-02 at 17 51 11" src="https://github.com/user-attachments/assets/789426e9-b232-49a7-a1cd-fd1fcb4a5af6" />

The notes in this repository say that you are "aiming for general release on 25th" - but I'm not sure 25th of what month and I'm also concerned by other comments saying that 2025.1 breaks this integration too?

**Comment by AndrewGWalsh** [comment]
Hello. I am using the 3.5.34 beta of this integration and am experiencing the same difficulties described above for my sockets but my lights appear to hold their names over reboots and to work correctly in automations and so on.

If there is anything I can do to help please say.

**Comment by Gazza-B** [comment]
Are you using the bigbadblunt integration? ( Lightwave RF)
If so then ditch it and go for the other one  (Lightwave Smart)

Gary Bridge MIPS, MIET, MAES.


GB Broadcast Radio Technology.
________________________________

**Reply by AndrewGWalsh** [reply]
No, I'm on the new one, Lightwave Smart - loads much faster even though I have 136 devices (209 reported by the new integration).

Kind thoughts,

Andrew

**Comment by Gazza-B** [comment]
I’m still on 3.5.31 works fine for me

Gary Bridge MIPS, MIET, MAES.


GB Broadcast Radio Technology.
________________________________

**Reply by AndrewGWalsh** [reply]
3.5.31 is much better, thank you for the pointer.

**Reply by ashy-87** [reply]
Does this version fix the resetting devices on reboot? And, Are multi-tap events seen on this version still?

**Reply by AndrewGWalsh** [reply]
My devices are stable and I haven't noticed any difficulty when rebooting. The multi-tap I don't know about I'm afraid.

**Comment by The-Latchkey-Kid** [comment]
This Lightwave Integration no longer works with the latest Home Assistant update :( .... Every light in my house is on a lightwave switch.  Can anyone recommend a working smart switch or let me know if this will ever get updated.

**Reply by TheAndyMac** [reply]
Have you tried 3.5.32 beta? I had some issues with production, then tried the latest beta (which did not work as expected) and, following a conversation about it, am now on 3.5.32 beta [see here](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/issues/35). This seems to be working although I use it only for simple integration between LightWave and Tado, using the LightWave Smart app for the automation itself.

**Reply by The-Latchkey-Kid** [reply]
Andrew, you absolute legend.  I thought I had all latest updates, but didn't realise, the beta isn't automatically updated like the others.  For anybody else having issues, this is what I did.  Go to hacs and press redownload on the 3 little dots on lightwave smart, (the one with the box).  Then choose 'a different version' and I selected 3.5.33-Beta.  Then restart.  Then go to devices and services and add the Lightwave smart integration, (same one as in the hacs),,, login and voila!!.......    All credit to @TheAndyMac for the solution.......

**Reply by TheAndyMac** [reply]
To be fair, it was [@nuttersrest](https://github.com/nuttersrest) who gave me the solution.

**Comment by ikb42** [comment]
Please see the latest beta, if you're able to test and provide feedback 👍  [v3.5.37](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v3.5.37-Beta)

**Reply by The-Latchkey-Kid** [reply]
Seems fine so far, if I find any issues I will report back.

**Comment by ikb42** [comment]
Hello all
Later this week I will promote the latest beta [v3.5.37](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v3.5.37-Beta) to general release.
Thumbs up if you agree, comment if not, thanks

**Comment by ikb42** [comment]
A new version is available, please give it a try - [v4.0.2-Beta](https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/releases/tag/v4.0.2-Beta)

Closing this discussion - please see latest https://github.com/LightwaveSmartHome/homeassistant-lightwave-smart/discussions/42
